Bug: Build error on Clang

### What happened?

```
[54/104] Building CXX object src/CMakeFiles/llama.dir/llama.cpp.o
FAILED: src/CMakeFiles/llama.dir/llama.cpp.o
/usr/bin/clang++ -DGGML_USE_CUDA -DLLAMA_BUILD -DLLAMA_SHARED -Dllama_EXPORTS -I/root/llama.cpp/src/. -I/root/llama.cpp/src/../include -I/root/llama.cpp/ggml/src/../include -O2 -O3 -DNDEBUG -fPIC -MD -MT src/CMakeFiles/llama.dir/llama.cpp.o -MF src/CMakeFiles/llama.dir/llama.cpp.o.d -o src/CMakeFiles/llama.dir/llama.cpp.o -c /root/llama.cpp/src/llama.cpp
/root/llama.cpp/src/llama.cpp:2660:11: error: call to implicitly-deleted default constructor of 'struct llama_grammar'
        , grammar()
          ^
/root/llama.cpp/src/./llama-grammar.h:9:32: note: default constructor of 'llama_grammar' is implicitly deleted because field 'rules' of const-qualified type 'const llama_grammar_rules' (aka 'const vector<vector<llama_grammar_element>>') would not be initialized
    const llama_grammar_rules  rules;
                               ^
/root/llama.cpp/src/llama.cpp:14045:38: warning: using floating point absolute value function 'fabs' when argument is of integer type [-Wabsolute-value]
                                f = -fabs(lctx.kv_self.cells[i].pos - pos);
                                     ^
/root/llama.cpp/src/llama.cpp:14045:38: note: use function 'std::abs' instead
                                f = -fabs(lctx.kv_self.cells[i].pos - pos);
                                     ^~~~
                                     std::abs
/root/llama.cpp/src/llama.cpp:14086:42: warning: using floating point absolute value function 'fabs' when argument is of integer type [-Wabsolute-value]
                                    f = -fabs(batch.pos[i] - batch.pos[j]);
                                         ^
/root/llama.cpp/src/llama.cpp:14086:42: note: use function 'std::abs' instead
                                    f = -fabs(batch.pos[i] - batch.pos[j]);
                                         ^~~~
                                         std::abs
2 warnings and 1 error generated.
```

Seems related to https://github.com/ggerganov/llama.cpp/pull/8508

### Name and Version

68504f0970db5a3602d176953690f503059906b1

### What operating system are you seeing the problem on?

_No response_

### Relevant log output

_No response_

--- a/src/llama-grammar.h
+++ b/src/llama-grammar.h
@@ -13,8 +13,6 @@ struct llama_grammar {
     llama_partial_utf8 partial_utf8;
 };
 
-struct llama_grammar * llama_get_grammar(struct llama_context * ctx);
-
 //
 // internal API
 //

--- a/src/llama.cpp
+++ b/src/llama.cpp
@@ -2657,7 +2657,6 @@ struct llama_context {
     llama_context(const llama_model & model)
         : model(model)
         , sampling(llama_n_vocab(&model))
-        , grammar()
         , t_start_us(model.t_start_us)
         , t_load_us(model.t_load_us) {}
 
@@ -2675,7 +2674,6 @@ struct llama_context {
 
     struct llama_cparams        cparams;
     struct llama_sampling       sampling;
-    struct llama_grammar        grammar;
     struct llama_kv_cache       kv_self;
     struct llama_control_vector cvec;
 
@@ -14048,7 +14046,7 @@ static void llama_set_inputs(llama_context & lctx, const llama_batch & batch) {
                             f = -INFINITY;
                         } else {
                             if (hparams.use_alibi) {
-                                f = -fabs(lctx.kv_self.cells[i].pos - pos);
+                                f = -std::abs(lctx.kv_self.cells[i].pos - pos);
                             } else {
                                 f = 0.0f;
                             }
@@ -14102,7 +14100,7 @@ static void llama_set_inputs(llama_context & lctx, const llama_batch & batch) {
                         for (int s = 0; s < batch.n_seq_id[i]; ++s) {
                             if (batch.seq_id[i][s] == seq_id) {
                                 if (hparams.use_alibi) {
-                                    f = -fabs(batch.pos[i] - batch.pos[j]);
+                                    f = -std::abs(batch.pos[i] - batch.pos[j]);
                                 } else {
                                     f = 0.0f;
                                 }
@@ -16833,10 +16831,6 @@ const struct llama_model * llama_get_model(const struct llama_context * ctx) {
 
 const struct llama_vocab * llama_get_vocab(const struct llama_context * ctx) {
     return &ctx->model.vocab;
-}
-
-struct llama_grammar * llama_get_grammar(struct llama_context * ctx) {
-    return &ctx->grammar;
 }
 
 uint32_t llama_n_ctx(const struct llama_context * ctx) {

